marshal.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package httprequest
+
+import (
+	"bytes"
+	"encoding"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"reflect"
+
+	"github.com/julienschmidt/httprouter"
+	"gopkg.in/errgo.v1"
+)
+
+// Marshal takes the input structure and creates an http request.
+//
+// See: Unmarshal for more details.
+//
+// For fields with a "path" item in the structural tag, the base uri must
+// contain a placeholder with its name.
+// Example:
+// For
+//    type Test struct {
+//	    username string `httprequest:"user,path"`
+//    }
+// ...the request url must contain a ":user" placeholder:
+//    http://localhost:8081/:user/files
+//
+// If a type does not implement the encoding.TextMarshaler fmt.Sprint will
+// be used to marshal its value.
+func Marshal(baseURL, method string, i interface{}) (*http.Request, error) {
```

Comment:
Let's mirror Unmarshal here and use "x" and "xv" instead of "i" and "input" (sorry matty :])